polynote-frontend/polynote/state/state_handler.ts

```diff
@@ -321,10 +328,12 @@ export class ObjectStateHandler<S extends object> extends Disposable implements
         this.update(keyUpdater(key, updateFn), updateSource, `${key.toString()}.` + (updateSubPath ?? ''))
     }
 
+    // returns a Disposable that will remove itself from the ObserverDict when disposed
```

Comment:
I would also add what this function does. e.g. "Adds an observer at a given path that is disposed when this handler is disposed. Returns a Disposable so callers can add introduce additional disposal criteria" (same as addPreObserverAt)